solidity/contracts/vault/TBTCVault.sol

```diff
@@ -55,7 +61,29 @@ contract TBTCVault {
     ///      for at least `amount`.
     /// @param amount Amount of TBTC to mint
     function mint(uint256 amount) external {
-        _mint(msg.sender, amount);
+        address minter = msg.sender;
+        require(
+            bank.balanceOf(minter) >= amount,
+            "Amount exceeds balance in the bank"
+        );
+        _mint(minter, amount);
+        bank.transferBalanceFrom(minter, address(this), amount);
+    }
+
+    /// @notice Mints the same amount of TBTC as the deposited amount for each
+    ///         depositor in the array. Can only be called by the Bank after the
+    ///         Bridge swept deposits and Bank increased balance for the
+    ///         vault.
+    /// @dev Fails if the lenght of `depositors` and `depositedAmounts`
+    ///      arrays is not the same. Fails if `depositors` array is empty.
```

Comment:
WIll not fail if `depositedAmounts` is longer than `depositors`. However, if we decide to validate only in `increaseBalanceAndCall`, this is no longer relevant. Also, `require(depositors.length != 0, "No depositors specified");` could be deleted in that case.

Reply:
https://github.com/keep-network/tbtc-v2/pull/94/commits/a4370e66f09b8da92eb7f14eb941b4c329a6baba

Reply:
There could be a scenario where some vault allows to pass empty arrays because it does not care about
who deposited, e.g. some fundraising contract. That's why we do not validate if depositors array is non-empty in `Bank` but we validate it in `TBTCVault` - for TBTC it matters.